code/modules/mining/equipment/mining_charges.dm

```diff
@@ -0,0 +1,213 @@
+		installed = TRUE
+		target.overlays += image_overlay
+
+/obj/item/grenade/plastic/miningcharge/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/detonator))
+		return
+	var/obj/item/detonator/detonator = I
+	if(!(src in detonator.bombs) && !timer_off)
+		detonator.bombs += src
+		timer_off = TRUE
+		to_chat(user, "<span class='notice'>You synchronize [src] to [I].</span>")
+		playsound(src, 'sound/machines/twobeep.ogg', 50)
+		detonator.update_icon()
+	else
+		to_chat(user, "<span class='warning'>[src] was already synchronized to a existing detonator!</span>")
+
+/obj/item/grenade/plastic/miningcharge/proc/detonate()
+	addtimer(CALLBACK(src, PROC_REF(prime)), 3 SECONDS)
+
+/obj/item/grenade/plastic/miningcharge/prime()
+	if(hacked) //try not to blow your fingers off
+		explode()
+		return
+	var/turf/simulated/mineral/location = get_turf(target)
+	var/datum/effect_system/smoke_spread/S = new
+	S.set_up(smoke_amount, 0, location, null)
+	S.start()
+	for(var/turf/simulated/mineral/rock in circlerangeturfs(location, boom_sizes[3]))
+		var/distance = get_dist_euclidian(location, rock)
+		if(distance <= boom_sizes[3])
+			rock.mineralAmt += 3 // if rock is going to get drilled, add bonus mineral amount
+			rock.gets_drilled()
+	for(var/mob/living/carbon/C in circlerange(location, boom_sizes[3]))
+		var/distance = get_dist_euclidian(location, C)
+		C.flash_eyes()
+		C.Weaken((boom_sizes[2] - distance) * 1 SECONDS) //1 second for how close you are to center if you're in range
+		C.AdjustDeaf((boom_sizes[3] - distance) * 5 SECONDS) //guaranteed deafness
+		var/obj/item/organ/internal/ears/ears = C.get_int_organ(/obj/item/organ/internal/ears)
+		if(istype(ears) && C.check_ear_prot() < HEARING_PROTECTION_MINOR) //headsets should be enough to avoid taking damage
+			ears.receive_damage((boom_sizes[3] - distance) * 2) //something like that i guess. Mega charge makes 12 damage to ears if nearby
+		to_chat(C, "<span class='userdanger'>You are knocked down by the power of the mining charge!</span>")
+	qdel(src)
+
+/obj/item/grenade/plastic/miningcharge/proc/explode() //c4 code
+	var/turf/location
+	if(target)
+		if(!QDELETED(target))
+			if(isturf(target))
+				location = get_turf(target)
+			else
+				location = get_atom_on_turf(target)
+			target.overlays -= image_overlay
+	else
+		location = get_atom_on_turf(src)
+	if(location)
+		explosion(location, boom_sizes[1], boom_sizes[2], boom_sizes[3], cause = src)
+		location.ex_act(2, target)
+	qdel(src)
+
+/obj/item/grenade/plastic/miningcharge/proc/override_safety()
+	hacked = TRUE
+	notify_admins = TRUE
+	boom_sizes[1] = round(boom_sizes[1] / 3)	//lesser - 0, normal - 0, mega - 1; c4 - 0
+	boom_sizes[2] = round(boom_sizes[2] / 3)	//lesser - 0, normal - 1, mega - 2; c4 - 0
+	boom_sizes[3] = round(boom_sizes[3] / 1.5)	//lesser - 2, normal - 3, mega - 5; c4 - 3
+
+/obj/item/grenade/plastic/miningcharge/deconstruct(disassembled = TRUE) //no gibbing a miner with pda bombs
```

Comment:
PDA bombs?

Reply:
discussed on discord, this means that we're overriding the superclass so damage to these grenades from other sources don't cause them to go off.

Reply:
Comment updated.